spec/unit/type/network_config_spec.rb

```diff
@@ -166,29 +178,59 @@
         end.to raise_error
       end
 
+      it 'fails on numeric values < 42' do
+        expect do
+          Puppet::Type.type(:network_config).new(name: 'yay', mtu: 41)
+        end.to raise_error
+      end
+
       it 'fails on zero' do
         expect do
           Puppet::Type.type(:network_config).new(name: 'yay', mtu: '0')
         end.to raise_error
       end
 
+      it 'fails on numeric zero' do
+        expect do
+          Puppet::Type.type(:network_config).new(name: 'yay', mtu: 0)
+        end.to raise_error
+      end
+
       it 'fails on values > 65536' do
         expect do
           Puppet::Type.type(:network_config).new(name: 'yay', mtu: '65537')
         end.to raise_error
       end
 
+      it 'fails on numeric values > 65536' do
+        expect do
+          Puppet::Type.type(:network_config).new(name: 'yay', mtu: 65_537)
+        end.to raise_error
+      end
+
       it 'fails on negative values' do
         expect do
           Puppet::Type.type(:network_config).new(name: 'yay', mtu: '-1500')
         end.to raise_error
       end
 
+      it 'fails on negative numbers' do
+        expect do
+          Puppet::Type.type(:network_config).new(name: 'yay', mtu: -1500)
+        end.to raise_error
+      end
+
       it 'fails on non-integer values' do
         expect do
           Puppet::Type.type(:network_config).new(name: 'yay', mtu: '1500.1')
         end.to raise_error
       end
+
+      it 'fails on numeric non-integer values' do
+        expect do
+          Puppet::Type.type(:network_config).new(name: 'yay', mtu: 1500.1)
+        end.to raise_error
+      end
```

Comment:
There are non numbered tests, they were the original tests - it assumed a String. I just added numerical tests.